lib/Grape.js

```diff
@@ -174,40 +166,53 @@ class Grape extends Events {
     this.get(hash, cb)
   }
 
-  announce (val, port, cb) {
+  announce (val, port = this.conf.dht_port, cb) {
     if (!_.isInteger(port)) {
       return cb(new Error('ERR_GRAPE_SERVICE_PORT'))
     }
 
     cb = cb || noop
     this.node.announce(
-      this.str2hex(val),
-      port || this.conf.dht_port,
+      this.str2hashbuf(val),
+      { port },
       cb
     )
   }
 
   lookup (val, cb) {
-    const ih = this.str2hex(val)
-
-    this.cbq0.push(ih, (err, data) => {
```

Comment:
what happens when there are 100 of the same lookups in a short time?

Reply:
definitely worth discussing: this would ultimately create 100 dht-rpc query streams (https://github.com/mafintosh/dht-rpc/blob/master/lib/query-stream.js) which use a concurrency setting that causes dht queries to be added to a pending queue when inflight queries (for the node on which lookup is called) exceed the concurrency limit. The inflight tracking and concurrency setting is on the actual node itself, rather than per query stream. Default concurrency is 16, so without configuration there won't be more than 16 inflight requests from this node to the DHT at any given time. 

Even given that, I'd definitely like to stress test these assumptions with minipass load testing once we have this merged